### ISSUE TITLE
macOS: speak() doesn't work and libraries aren't copied over

### ISSUE THREAD

**Issue by patricus3** [samtupy/nvgt#16]
hi.
I tried to use nvgt and wrote some code, just for testing purposes, everything compiles fine, I can't run it until I put necessary libraries into the folder though, and also there's an issue with the speak() function.

**Comment by samtupy** [comment]
Hi, our voice over implementation requires that a window be shown in order to produce speech. This is a limitation of Apple's API and not something we can easily work around in the short term. Can you please try showing a window first and make sure it displays before speaking text?

**Comment by patricus3** [comment]
don't use voiceover for speech output, queueing isn't working there because apple is apple

**Comment by patricus3** [comment]
I don't know C/C++ so I can't help.
and, also, I tried to make a window, and it didn't worked, like, at all.

**Comment by samtupy** [comment]
That is not helpful without extra information, an example, something. Creating a window on macOS has worked for everybody thus far. What didn't work, creating a window or speaking with VO? Was the window showing on screen by the time the text spoke?
If you wrote something like:
```
void main() {
show_window("test");
screen_reader_speak("hello", true);
}
```
This of course won't work because the window can't even show before the application exits. Thus, are you speaking text from within the context of a game E. showing a window and speaking in response to a keypress or something?

**Comment by patricus3** [comment]
I also add wait to wait 5 ms, I use it in the main context, without any key presses

**Comment by samtupy** [comment]
Yeah that's why, the VO speech really does only work in a game context. You need to actually open a loop and wait for the window to focus, then and only then will VO speak text. The window will absolutely not appear within 5ms after the show_window call, not even 100ms after. You need to create a loop and wait for the user to press escape, then you can speak. As an example, right now if I run stw on macos, the main menu text is not spoken because the main menu launches before VO registers the existence of the new window. Once VO tells me that I'm in the Survive the Wild window though, I can then safely up and down arrow through my main menu and get consistent speech.

**Comment by samtupy** [comment]
Here, does this break for you in any way? I apologize for any compilation errors as I wrote the code specifically for the issue here.
```
void main() {
	show_window("test");
	while(!key_pressed(KEY_ESCAPE)) { // Does not need to be escape, can be anything
		wait(5);
		if (key_pressed(KEY_SPACE)) screen_reader_speak("hello", true);
	}
}
```

**Comment by patricus3** [comment]
works, but sometimes you want it to say something on game start, and, well...

**Comment by patricus3** [comment]
ah ah ok
also, I see that there's dynamic menu include that isn't documented yet,
and, you mentioned stw on macOS, I don't see it on the website.
could I have a link to test it?

**Comment by peter1384** [comment]
@patricus3 LOL, I think that @samtupy said stw for MakOS as an example for you to understand the point, There is currently no STW for MakOs.

**Comment by samtupy** [comment]
We can talk about anything stw beta related outside of NVGT's github issues as: 1, that could quickly get off topic and 2, if you ask for the stw beta and I give it to you, others will do the same and soon loads of people will be asking for a beta that I have obviously chosen to not release yet. Thanks for the interest though, I'm certainly excited to get to a release ready point with that!

**Comment by samtupy** [comment]
I think we forgot to mark this as resolved when NVGT's bundling facility got introduced, I believe this is very much no longer an issue, please feel free to create a new one if not.
